experiments/run-big-bench.py

```diff
@@ -207,8 +206,14 @@ def create_and_train_hnsw_index(
     dataset_size: int,
     ef_construction: int,
     max_edges_per_node: int,
-    num_threads,
-) -> hnswlib.Index:
+    num_threads: int,
+    base_layer_filename: Optional[str] = None,
+) -> Union[hnswlib.Index, None]:
+    """
+    Kind of messy to return either a hnswlib.Index or None, but we are doing this so that
+    when we use the HNSW base layer to construct a Flatnav index, we don't ever have the 
+    two indices in memory at the same time. This should help with memory usage.
```

Comment:
If we don't return the HNSW index, it should be deallocated when the function exits since it's scope is limited to this function (in case we return None).